lib/job.js

```diff
@@ -393,8 +393,19 @@ Job.prototype._saveAttempt = function(){
 /**
 */
 Job.fromData = function(queue, jobId, data){
-  var job = new Job(queue, JSON.parse(data.data), JSON.parse(data.opts));
-  job.jobId = jobId;
+  try {
+    var job = new Job(queue, JSON.parse(data.data), JSON.parse(data.opts));
+    job.jobId = jobId;
+  }
+  catch (err) {
+    //Job unable to be parsed. Remove it.
+    var job = new Job(queue, JSON.parse(data.data), JSON.parse(data.opts));
```

Comment:
The above throws because `JSON.parse` attempts to parse undefined. Wouldn't this throw the error again?